lib/pages/forward/forward_view.dart

```diff
@@ -31,32 +41,66 @@ class _ForwardViewState extends State<ForwardView> {
     });
   }
 
-  void forwardAction(BuildContext context) async {
-    final rooms = widget.controller.filteredRoomsForAll;
-    final room = rooms.firstWhere((element) => element.id == widget.controller.selectedEvents.first);
+  void _toggleContactChats() {
+    setState(() {
+      isShowContactChats = !isShowContactChats;
+    });
+  }
+
+  void _forwardRecentlyChatAction(List<Room> rooms) async {
+    final room = rooms.firstWhere((element) => element.id == widget.controller.selectedEvents.first.id);
+    _sendForwardMessageAction(room);
+  }
+
+  void _forwardContactsChatAction(List<Room> rooms) async {
+    final roomId = await Matrix.of(context).client.startDirectChat(
+      widget.controller.selectedEvents.first.id.toTomMatrixId()
+    );
+
+    final hasRoom = rooms.any((element) => element.id == roomId);
+    if (hasRoom) {
+      final room = rooms.firstWhere((element) => element.id == roomId);
+      _sendForwardMessageAction(room);
+    } else {
+      await Matrix.of(context).client.getRoomById(roomId)!.leave();
```

Comment:
what is it? why `leave`?

Reply:
My point: 

1. When I selected one item and click forward then I got `roomId`.
2. If I call `Matrix.of(context).client.startDirectChat` => `Returns an existing direct room ID with this user or creates a new one.`

```
 final roomId = await Matrix.of(context).client.startDirectChat(
      widget.controller.selectedEvents.first.id.toTomMatrixId()
    );
```

3. If `roomId` existed in recent chat  => forward msg else => remove `roomId`.

I think we need to discuss with team about this